students/david_russo/lesson4/mailroom_lesson04.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+from sys import exit
+import random
+
+# construct list of donors with a frozen set, to later be used as keys in the dictionary
+donor_names = frozenset(("Daniel Ocean", "Rusty Ryan", "Linus Caldwell", "Ruben Tishkoff", "Basher Tarr"))
+
+# create an empty dictionary to be populated later
+donors = {}
+
+# populate dictionary with a random number of random values per key using the setdefault method 
+for donor in donor_names:
+    donors.setdefault(donor, random.sample(range(1, 50000), random.randint(1, 5)))
+
+# define a function to prompt the user for their desired action
+def prompt_user():
+    print("Choose an action. ")
+    print()
+    print("1 - Send a Thank You")
+    print("2 - Create a Report")
+    print("3 - Send letters to everyone")
+    print("4 - Quit")
+    response = int(input(" > "))
+    while response not in (1, 2, 3, 4):
+        response = int(input("Please choose one of 1, 2, 3, or 4: > "))
+    return response
+
+
+# define a function to send a thank you card
+def send_a_thank_you():
+    # get a list of unique donors
+    unique_donors = donors.keys()
+    # prompt the user for a donor
+    donor = input("To whom would you like to send a thank you? > ")
+    # list the possible candidates within the donors list    
+    if donor == "list":
+        # print the donors in the dict
+        print(', '.join(list(unique_donors)))
+        donor = input("To whom would you like to send a thank you? > ")
+    # prompt the user for a donation amount
+    donation_amount = int(input("How much would you like to donate? > "))
+    # add the donor and the donation amount to the donors list
+    donors.setdefault(donor, donation_amount)
+    # Print the thank you note
+    print("Dear {:s}, \n Thank you for your generous donation of ${:.2f}. \n".format(donor, donation_amount),
+          "Best, \n The Donation Foundation.")
```

Comment:
This function doesn't appear to add to the donor's total amount.  Take another look and see what you might have missed.
Reproduce steps:
```
1. Select Create Report
2. Note the total money for the donors
3. Select Send Thank you
4. Enter a Donor name and enter a donor amount of 500
5. Select Create Report

bug: Donor amounts remain the same as in step one
expected: Donor amounts to increment on the donation of $500
```